core/finality-grandpa/src/environment.rs

```diff
@@ -455,6 +468,9 @@ where
 	RA: 'static + Send + Sync,
 	SC: SelectChain<Block> + 'static,
 	NumberFor<Block>: BlockNumberOps,
+	T: PoolApi,
+	<T as PoolApi>::Api: txpool::ChainApi<Block=Block>,
```

Comment:
I'm not convinced that we need to tightly couple to the transaction pool here. Could we introduce a trait `ReportHook` that has an implementation for `T: PoolApi`?